overlord/snapshotstate/backend/backend.go

```diff
@@ -549,23 +609,16 @@ func (t *importTransaction) Cancel() error {
 	if err != nil {
 		return err
 	}
-	var errs []error
+	errs := NewMultiError("cannot cancel import for set id %d", t.id)
```

Comment:
constructing errors up front this way reads also strangely, it puts emphasis on the wrong thing imo

Reply:
Indeed, it's a bit clunky. Something like this is needed to get the nesting right,, maybe move it ?
```
index 1628daa933..a798d39252 100644
--- a/overlord/snapshotstate/backend/backend.go
+++ b/overlord/snapshotstate/backend/backend.go
@@ -607,7 +607,7 @@ func (t *importTransaction) Cancel() error {
        if err != nil {
                return err
        }
-       errs := newMultiError("cannot cancel import for set id %d", t.id)
+       errs := newMultiError()
        for _, p := range inProgressImports {
                if err := os.Remove(p); err != nil {
                        errs.Append(err)
@@ -616,7 +616,10 @@ func (t *importTransaction) Cancel() error {
        if err := t.unlock(); err != nil {
                errs.Append(err)
        }
-       return errs.Err()
+       if err := errs.Err(""); err != nil {
+               return errs.Err("cannot cancel import for set id %d", t.id)
+       }
+       return nil
 }
 
 // Commit will commit a given transaction
```
but also feels a bit clunky. Ideas welcome!

Reply:
maybe I'm missing something but the easiest change is to keep the old "var errs []error" declaration, not have Append but have:

newMultiError(errs []error, headerFmt string, v ...interface{}) error

or a bit more natural

newMultiError(header string, errs []error) error

and Sprintf to make the header?

Reply:
Thanks, I updated the code to follow this pattern now.